Doesn't compile with Mono master (4.9+)

Please provide a succinct description of the issue.

Build error when compiling with Mono 4.9

#### Repro steps

Provide the steps required to reproduce the problem

1. Install Mono 4.9 https://bosstoragemirror.blob.core.windows.net/wrench/mono-master/c1/c1c09bc78d33dda29a4e53aa833f8ed09bbadaf2/MonoFramework-MDK-4.9.4.26.macos10.xamarin.universal.pkg

2. Build this repo
```
./autogen.sh --prefix=/Library/Frameworks/Mono.framework/Versions/4.9.4
make
```
#### Expected behavior

It should build

#### Actual behavior

```
XBuild Engine Version 14.0
Mono, Version 4.9.4.0
Copyright (C) 2005-2013 Various Mono authors

Build started 3/21/2017 1:12:44 PM.
__________________________________________________
Project "/Users/jason/src/fsharp/src/fsharp/FSharp.Build-proto/FSharp.Build-proto.fsproj" (default target(s)):
	Target CallFsSrGen:
		Created directory "obj//net40/"
		Executing: mono /Users/jason/src/fsharp/packages/FSharp.Compiler.Tools.4.0.1.21/tools/fsi.exe --exec "/Users/jason/src/fsharp/src/scripts/fssrgen.fsx" "/Users/jason/src/fsharp/src/fsharp/FSharp.Build/FSBuild.txt" "obj//net40/FSBuild.fs"  "obj//net40/FSBuild.resx"
		error FS0193: Specified method is not supported.
/Users/jason/src/fsharp/src/scripts/fssrgen.targets: error : Command 'mono /Users/jason/src/fsharp/packages/FSharp.Compiler.Tools.4.0.1.21/tools/fsi.exe --exec "/Users/jason/src/fsharp/src/scripts/fssrgen.fsx" "/Users/jason/src/fsharp/src/fsharp/FSharp.Build/FSBuild.txt" "obj//net40/FSBuild.fs"  "obj//net40/FSBuild.resx" ' exited with code: 1.
	Task "Exec" execution -- FAILED
	Done building target "CallFsSrGen" in project "/Users/jason/src/fsharp/src/fsharp/FSharp.Build-proto/FSharp.Build-proto.fsproj".-- FAILED
Done building project "/Users/jason/src/fsharp/src/fsharp/FSharp.Build-proto/FSharp.Build-proto.fsproj".-- FAILED

Build FAILED.
Errors:

/Users/jason/src/fsharp/src/fsharp/FSharp.Build-proto/FSharp.Build-proto.fsproj (default targets) ->
/Users/jason/src/fsharp/src/scripts/fssrgen.targets (CallFsSrGen target) ->

	/Users/jason/src/fsharp/src/scripts/fssrgen.targets: error : Command 'mono /Users/jason/src/fsharp/packages/FSharp.Compiler.Tools.4.0.1.21/tools/fsi.exe --exec "/Users/jason/src/fsharp/src/scripts/fssrgen.fsx" "/Users/jason/src/fsharp/src/fsharp/FSharp.Build/FSBuild.txt" "obj//net40/FSBuild.fs"  "obj//net40/FSBuild.resx" ' exited with code: 1.

	 0 Warning(s)
	 1 Error(s)

Time Elapsed 00:00:02.9666400
make[3]: *** [build-proto] Error 1
make[2]: *** [build-proto] Error 2
make[1]: *** [all] Error 2
make: *** [all] Error 2
```

#### Known workarounds

None

#### Related information 

Provide any related information 

* Operating system
* Branch
* .NET Runtime, CoreCLR or Mono Version
* Editing Tools (e.g. Visual Studio Version)
* Links to F# RFCs or entries on http://fslang.uservoice.com
* Links to performance testing scripts
* Indications of severity

--- a/src/absil/ilreflect.fs
+++ b/src/absil/ilreflect.fs
@@ -616,16 +616,18 @@ let convFieldInit x =
 // it isn't we resort to this technique...
 let TypeBuilderInstantiationT = 
     let ty = 
+        Type.GetType("System.Reflection.Emit.TypeBuilderInstantiation")
 #if ENABLE_MONO_SUPPORT
-        if runningOnMono then 
+    let ty =
+        if runningOnMono && (isNull ty) then
             Type.GetType("System.Reflection.MonoGenericClass")
         else
+            ty
 #endif
-            Type.GetType("System.Reflection.Emit.TypeBuilderInstantiation")
     assert (not (isNull ty))
     ty
 
-let typeIsNotQueryable (typ : Type) =
+let typeIsNotQueryable (typ : Type) = 
 #if FX_RESHAPED_REFLECTION
     let typ = typ.GetTypeInfo()
 #endif

--- a/src/scripts/fssrgen.fsx
+++ b/src/scripts/fssrgen.fsx
@@ -1,5 +1,4 @@
 ﻿// Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
-
 let PrintErr(filename, line, msg) =
     printfn "%s(%d): error : %s" filename line msg
         
@@ -9,7 +8,7 @@ let Err(filename, line, msg) =
     printfn "# comment"
     printfn "ident,\"string\""
     printfn "errNum,ident,\"string\""
-    failwithf "there were errors in the file '%s'" filename
+    failwith (sprintf "there were errors in the file '%s'" filename)
 
 let xmlBoilerPlateString = @"<?xml version=""1.0"" encoding=""utf-8""?>
 <root>

